pulser-core/pulser/channels/eom.py

```diff
@@ -96,27 +122,35 @@ def _to_abstract_repr(self) -> dict[str, Any]:
 
 
 @dataclass(frozen=True)
-class RydbergEOM(BaseEOM):
+class _RydbergEOM:
+    limiting_beam: RydbergBeam
+    max_limiting_amp: float  # rad/µs
+    intermediate_detuning: float  # rad/µs
+    controlled_beams: tuple[RydbergBeam, ...]
+
+
+@dataclass(frozen=True)
+class _RydbergEOMDefaults:
+    multiple_beam_control: bool = True
+
+
+@dataclass(frozen=True)
+class RydbergEOM(_RydbergEOMDefaults, BaseEOM, _RydbergEOM):
```

Comment:
You are right, it will change the order of the arguments, but I checked everywhere and we are always defining `mod_bandwidth` with a keyword so I took a chance.
Your suggestion is not ideal because `RydbergEOM` won't be a subclass of `BaseEOM` anymore

Reply:
Oh yes that's true, well let's take that chance :)